backend/lib/piece-indexer.js

```diff
@@ -0,0 +1,48 @@
+import { fetchPayloadCid } from './pix-service/service.js'
+import { parseDeals, storeActiveDeals } from './deal-observer.js'
+
+/** @import {Queryable} from '@filecoin-station/deal-observer-db' */
+/** @import { Static } from '@sinclair/typebox' */
+/** @import { ActiveDealDbEntry } from '@filecoin-station/deal-observer-db/lib/types.js' */
+
+/**
+ * @param {import("@filecoin-station/deal-observer-db").Queryable} pgPool
+ * @param {function} makeRpcRequest
+ * @param {function} pixRequest
+ * @returns {Promise<void>}
+ */
+export async function updatePayloadCids (pgPool, makeRpcRequest, activeDeals, pixRequest) {
+  const updatedDeals = []
+  for (const deal of activeDeals) {
+    const payloadCid = await fetchPayloadCid(deal.miner_id, deal.piece_cid, makeRpcRequest, pixRequest)
```

Comment:
There may be multiple deals with the same combination of `(miner_id, piece_cid)` values, that's how the currently proposed UNIQUE constraint works.

From what I've seen in fil-deal-ingester, it's very very likely there will be many such duplicates.

I created a follow-up issue to improve a more efficient handling of such duplicates, see https://github.com/CheckerNetwork/deal-observer/issues/59